app/controllers/concerns/two_factor_authenticatable.rb

```diff
@@ -6,6 +6,8 @@ module TwoFactorAuthenticatable
     before_action :handle_two_factor_authentication
     before_action :check_already_authenticated
     before_action :reset_attempt_count_if_user_no_longer_locked_out, only: :create
+
+    before_action :apply_secure_headers_override, only: :show
   end
```

Comment:
The reason this is here, is because the 2FA pages are the last forms to be shown before the redirect to the OIDC client (the user clicks "submit" with their 2fa and then gets redirected outside the app, so since this is the form, the CSP headers need to be on this page).

Alternatively, we could do this at the `ApplicationController` maybe? So that if the flow changes somehow, we'll always set the CSP headers correctly?